### ISSUE TITLE
Is HtmlUnit slower then crawer?

### ISSUE THREAD

**Issue by mikebgrep** [HtmlUnit/htmlunit#589]
I didn't see a discussion tab and I need to say I enjoy the library but do is faster or same speed as Web Crawler like jsoup or is slower?

**Comment by rbri** [comment]
It depends :-)

The major difference to jsoup is the support for javascript (and css). If the content you like to crawl is (partially) created by javascript (or the page you like to crawl is only reachable via pages requiring javascript for doing the navigation) then you have no option.

If you simply like to do a get requests and parse the returning html then (hopefully) the speed is comparable (there are options to disable e.g. javascript and css).

In some cases HtmlUnit is also faster than real browsers (because there is no page rendering) but the js execution is in most cases slower than in real browsers.

And of course i'm open for suggestions and improvements....

**Comment by mikebgrep** [comment]
Yes I am disabled the javascript and css and I parse only the response its work well but in some pages it's take more time.I will try to optimaze it.But was just curious if the speed is same or closer in this case I prefer to use HtmlUnit.

**Comment by rbri** [comment]
Any chance to provide the url of the slow pages? Maybe via private email?
Will have a look....

**Comment by mikebgrep** [comment]
The problem is because there a table with elements and when you select  next results it load them on top of the rest so the rows (li) can be up to 500 or more and go for up to 10 seconds to parse them extract the text from child nodes etc. and the java code to return the end result.I will try with indexing of the parent node and will see.Keep you posted.

**Comment by rbri** [comment]
@mikebgrep 
i made a change in the dom creation of tables (https://github.com/HtmlUnit/htmlunit/commit/cbd6ca82cd8420ab82e6795c99cbd0537af4e11e) that will have a huge effect in some situations - maybe you can switch to the 3.2.0 release and check if this has some (hopefully positive) effect in your case

**Comment by mikebgrep** [comment]
, will update to the new release when I have a chance.Thank you
